### ISSUE TITLE
Traits properties conflicts resolution False-Positive

### ISSUE THREAD

**Issue by 9ae8sdf76** [kalessil/phpinspectionsea#1232]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Extended) v3.0.11 |
| Language level | PHP 7.2 |

Current behaviour
--
```php
trait Bar {
    /**
     * @ORM\Column(name="bar", type="integer", options={"unsigned": true})
     */
    private $bar;

    public function getBar(): int
    {
        return $this->bar;
    }

    public function setBar(int $bar): void
    {
        $this->bar = $bar;
    }
}

class Foo {
    use Bar;

    /**
     * @ORM\Column(name="bar", type="integer", options={"unsigned": true})
     * @ORM\Id()
     * @ORM\GeneratedValue(strategy="IDENTITY")
     */
    private $bar;
}
```

![Screen Shot 2019-03-11 at 1 00 06 PM](https://user-images.githubusercontent.com/2528463/54142226-a9703200-43fd-11e9-91d2-17737bb61f1f.png)

Defining a trait with a property, and then overriding the property in a class using the trait with the same visibility and initial value will produce the inspection warning saying: 

> ''Foo' and 'Bar' define the same property ($pidm). less... (⌘F1) 
> Inspection info: Documentation can be found [here](https://secure.php.net/manual/en/language.oop5.traits.php#language.oop5.interfaces.examples.ex1)

Expected behaviour
--
No warning. I am compartmentalizing some of my more common doctrine annotations, and there are some cases where I need to replace the doctrine annotations (e.g., make a field part of a primary key). By declaring the property in my concrete class with my new annotations, doctrine will do the Right Thing (tm). However, EA Inspections (Extended) will flag those edge cases as wrong. I'd really like to use the inspection due to the fact that changing the visibility or default type would result in a fatal error. 

In the paragraph right above the "Example #12 Conflict Resolution" section of the linked documentation it says that the class cannot declare a property from a trait **unless** the visibility (private in my case) and default value (null or none in my case) are the same. Additionally, it also says that before PHP 7.0.0, declaring a property with the same visibility and default value will generate an E_STRICT notice.

Environment details
---

```
PhpStorm 2018.3.5
Build #PS-183.5912.26, built on March 7, 2019
...
JRE: 1.8.0_152-release-1343-b28 x86_64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
macOS 10.14.3
```

**Comment by kalessil** [comment]
Fixed!

**Comment by 9ae8sdf76** [comment]
Thank you for the quick turnaround! 

Looking over what the fix is, it looks like the warning will still appear if I override a trait property without annotating it: 

```php
trait Foo {
    public $bar;
}

class Bar {
    use Foo;

    /** @Id */
    protected $bar = 'foo';
}
```

The above example is invalid, but since I put the annotation there the warning would not get displayed. Thoughts?

**Comment by kalessil** [comment]
IDE will report this case (seems this a native inspection by PhpStorm).
